src/igvfd/tests/test_schema_reference_file.py

```diff
@@ -33,3 +33,87 @@ def test_transcriptome_annotation_dependency(testapp, reference_file):
         }, expect_errors=True
     )
     assert res.status_code == 422
+
+
+def test_schema_reference_file_validation_error_detail_only_allowed_when_upload_status_is_invalidated(testapp, reference_file):
+    # Given a file with upload_status: pending:
+    testapp.patch_json(reference_file['@id'], {'upload_status': 'pending'}, status=200)
+    # When validation_error_details added, then patch fails:
+    testapp.patch_json(
+        reference_file['@id'],
+        {
+            'validation_error_detail': 'some problem with validating the file'
+        },
+        status=422
+    )
+    # Given a file with upload_status: invalidated:
+    testapp.patch_json(reference_file['@id'], {'upload_status': 'invalidated'}, status=200)
+    # When validation_error_details added, then patch works:
+    testapp.patch_json(
+        reference_file['@id'],
+        {
+            'validation_error_detail': 'some problem with validating the file'
+        },
+        status=200
+    )
+
+
+def test_schema_reference_file_must_remove_validation_error_detail_before_upload_status_changed_to_pending(testapp, reference_file):
+    # Given a file with upload_status: invalidated and validation_error_detail:
+    testapp.patch_json(
+        reference_file['@id'],
+        {
+            'upload_status': 'invalidated',
+            'validation_error_detail': 'some problem with validating the file'
+        },
+        status=200
+    )
+    # When upload_status changed to pending, then patch fails:
+    testapp.patch_json(reference_file['@id'], {'upload_status': 'pending'}, status=422)
+    # Given a file with validation_error_detail removed:
+    r = testapp.get(reference_file['@id'] + '@@raw').json
+    r.pop('validation_error_detail')
+    testapp.put_json(reference_file['@id'], r, status=200)
+    # When upload_status changed to pending, then patch works:
+    testapp.patch_json(reference_file['@id'], {'upload_status': 'pending'}, status=200)
+
+
+def test_schema_reference_file_regeneratiing_upload_credentials_leads_to_valid_schema(testapp, reference_file):
```

Comment:
typo in the name of the function regeneratiing -> regenerating

Reply:
Fixed.